01-front-end-fundamentals/git-ssh/readme.md

```diff
@@ -0,0 +1,161 @@
+
+## Step 0: Checking for existing Keys
+First, we need to check for existing SSH keys on your computer. Open Terminal and enter:
+
+  ```
+  $ ls -al ~/.ssh
+  # Lists the files in your .ssh directory, if they exist
+  ```
+
+If you already have a key, skip to step 2. Otherwise proceed to step 1.
+
+
+## Step 1: Generate a new SSH Key
+1. With Terminal still open, copy and paste the text below. Make sure you substitute in your GitHub email address.
+
+  ```
+  $ ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+  # Creates a new ssh key, using the provided email as a label
+  # Generating public/private rsa key pair.
+  ```
+
+2. We strongly suggest keeping the default settings as they are, so when you're prompted to "Enter a file in which to save the key", just press Enter to continue.
+
+  ```
+  Enter file in which to save the key (/Users/you/.ssh/id_rsa): [Press enter]
+  ```
+
+3. You'll be asked to enter a passphrase.
+
+  ```
+  Enter passphrase (empty for no passphrase): [Type a passphrase]
+  # Enter same passphrase again: [Type passphrase again]
+  ```
+
+4. After you enter a passphrase, you'll be given the fingerprint, or id, of your SSH key. It will look something like this:
+
+  ```
+  Your identification has been saved in /Users/you/.ssh/id_rsa.
+  # Your public key has been saved in /Users/you/.ssh/id_rsa.pub.
+  # The key fingerprint is:
+  # 01:0f:f4:3b:ca:85:d6:17:a1:7d:f0:68:9d:f0:a2:db your_email@example.com
+  ```
+Great now that we have created a new key, we need to add it to the ssh-agent so we can easily access it.
+
+## Step 3: Add Your Key to the SSH-Agent
+To configure the ssh-agent program to use your SSH key:
+1. Ensure ssh-agent is enabled:
+
+  ```
+  # start the ssh-agent in the background
+  $ eval "$(ssh-agent -s)"
+  # Agent pid 59566
+  ```
+
+2. Add your SSH key to the ssh-agent:
+
+  ```
+  $ ssh-add ~/.ssh/id_rsa
+  ```
+
+*Note*: If you didn't generate a new SSH key in Step 1, and used an existing SSH key instead, you will need to replace `id_rsa` in the above command with the name of your existing private key file.
+
+## Step 3: Add Your SSH Key to your Github Account
+To configure your GitHub account to use your SSH key:
+
+Copy the SSH key to your clipboard with:
+
+  ```
+  $ pbcopy < ~/.ssh/id_rsa.pub
+  # Copies the contents of the id_rsa.pub file to your clipboard
+  ```
+
+*Note*: It's important to copy the key exactly without adding newlines or whitespace.
+
+Keep in mind that your key may also be named id_dsa.pub, id_ecdsa.pub or id_ed25519.pub, in which case you must change the filename in the above command
+
+Then in Github:
+
+1. In the top right corner of any page, click your  profile photo, then click Settings.
+2. In the user settings sidebar, click SSH keys.
+3. Click Add SSH key.
+4. In the Title field, add a descriptive label for the key.
```

Comment:
"add a descriptive label that uniquely identifies the computer you're currently using, e.g. `Jesse's MacBook Air`"